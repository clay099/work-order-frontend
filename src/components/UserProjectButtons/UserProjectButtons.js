import React from "react";
import useToggle from "../../hooks/useToggle";
import { Dialog, DialogTitle, Button, DialogActions } from "@material-ui/core/";
import { useSelector, useDispatch } from "react-redux";
import UpdateProject from "../UpdateProject/UpdateProject";
import { deleteProjectFromAPI } from "../../actions/projects";
import { useHistory } from "react-router-dom";

const UserProjectButtons = ({ status, id, token }) => {
	const {
		setTrue: handleOpenUpdate,
		setFalse: handleCloseUpdate,
		toggle: openUpdate,
	} = useToggle(false);

	const {
		setTrue: handleOpenDelete,
		setFalse: handleCloseDelete,
		toggle: openDelete,
	} = useToggle(false);

	const dispatch = useDispatch();
	let history = useHistory();

	const handleDelete = () => {
		dispatch(
			deleteProjectFromAPI({
				token,
				projectId: id,
			})
		);
		history.push(`/user`);
	};

	return (
		<>
			<Button
				variant="contained"
				color="primary"
				onClick={handleOpenUpdate}
			>
				Update Project
			</Button>
			<UpdateProject handleClose={handleCloseUpdate} open={openUpdate} />

			{status === "auction" ? (
				<>
					<Button
						variant="contained"
						color="secondary"
						onClick={handleOpenDelete}
					>
						Delete Project
					</Button>
					<Dialog onClose={handleCloseDelete} open={openDelete}>
						<DialogTitle id="delete-project">
							Are You Suer You Want To Delete This Project?
						</DialogTitle>
						<DialogActions>
							<Button
								variant="outlined"
								color="secondary"
								onClick={handleCloseDelete}
							>
								Cancel
							</Button>
							<Button
								variant="contained"
								color="secondary"
								onClick={handleDelete}
							>
								Delete Project
							</Button>
						</DialogActions>
					</Dialog>
				</>
			) : null}
		</>
	);
};

export default UserProjectButtons;
