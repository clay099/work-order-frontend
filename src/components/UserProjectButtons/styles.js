import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
	userProjectButtons: {
		marginTop: theme.spacing(4),
	},
}));

export default useStyles;
