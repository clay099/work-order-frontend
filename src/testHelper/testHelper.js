// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
