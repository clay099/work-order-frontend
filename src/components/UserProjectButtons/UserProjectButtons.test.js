import React from "react";
import UserProjectButtons from "./UserProjectButtons";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const UserProjectButtonsParams = {
	status: "auction",
	id: 1,
	token: "token",
};

describe("UserProjectButtons Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<UserProjectButtons {...UserProjectButtonsParams} />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(
			<UserProjectButtons {...UserProjectButtonsParams} />
		);
		expect(asFragment).toMatchSnapshot();
	});

	test("should have relevant buttons for auction stage", () => {
		const { getByText } = render(
			<UserProjectButtons {...UserProjectButtonsParams} />
		);

		const button = getByText("Delete Project");
		expect(button).toBeInTheDocument();
	});

	test("should have relevant buttons for progressing stage", () => {
		UserProjectButtonsParams.status = "progressing";
		const { getByText } = render(
			<UserProjectButtons {...UserProjectButtonsParams} />
		);

		const button = getByText("Mark Project As Completed");
		expect(button).toBeInTheDocument();
	});

	test("should have relevant buttons for completed stage", () => {
		UserProjectButtonsParams.status = "completed";
		const { getByText } = render(
			<UserProjectButtons {...UserProjectButtonsParams} />
		);

		const button = getByText("Review Project");
		expect(button).toBeInTheDocument();
	});
});
