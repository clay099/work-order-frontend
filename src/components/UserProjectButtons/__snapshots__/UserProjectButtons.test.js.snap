// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProjectButtons Component matches snapshot 1`] = `[Function]`;
